libs/core/langchain_core/example_selectors/semantic_similarity.py

```diff
@@ -37,34 +38,59 @@ class Config:
         extra = Extra.forbid
         arbitrary_types_allowed = True
 
+    @staticmethod
+    def _example_to_text(
+        example: Dict[str, str], input_keys: Optional[List[str]]
+    ) -> str:
+        if input_keys:
+            return " ".join(sorted_values({key: example[key] for key in input_keys}))
+        else:
+            return " ".join(sorted_values(example))
+
+    def _documents_to_examples(self, documents: List[Document]) -> List[dict]:
```

Comment:
One question -- why a static method for example to text vs. _documents_to_examples being a regular method

Are implementation providers going to need to override documents to examples based on some state available in self?